### PR TITLE
Blog home page, list blogs, details blog, users show

### PR DESCRIPTION
**Pull request by aadityaa-cuelogic**
Show recent blogs on home page
Blog listing should show : blog title, blog post upto 100 char, author name, category of blog, Image if any
Show menu options : Home, Category, Login if unregistered user, Search
Home click should reset to home page
Category click should show list of categories
Blogs content should be hidden if more than 100 characters
Click on blog title should redirect to details post page

### CODE REVIEW

File: blog/blogs/forms.py [hunk 1]
```diff
@@ -3,8 +3,43 @@
...
+
+    def clean_title(self):
+        try:
+            post = Post.objects.get(title__iexact=self.cleaned_data['title'])
+        except Post.DoesNotExist:
+            return self.cleaned_data['title']
+        raise forms.ValidationError(
+            _("This post title already exits. Please try something else!"))
+
+    # def clean_slug(self):
```
Comment (ananthu-nair707) [inline blog/blogs/forms.py]: If not needed, remove commented code.

File: blog/blogs/urls.py [hunk 2]
```diff
@@ -3,4 +3,7 @@
 
 urlpatterns = [
     url(r'^$', views.home, name="home"),
+    url(r'^blog/(?P<slug>[\w-]+)/$', views.detailPost, name='blog_detail'),
+    url(r'^user/newpost/$', views.createPost, name='create_post'),
+    url(r'^user/newpost/success$', views.createPostSuccess, name='create_post_success'),
```
Comment (ananthu-nair707) [inline blog/blogs/urls.py]: 80 chars.

File: blog/blogs/views.py [hunk 3]
```diff
@@ -2,15 +2,63 @@
...
+# home page method
+# @login_required(login_url="login/")
 def home(request):
-    return render(request, "home.html")
+    latest_blog_post = Post.objects.order_by('-created_on')[:5]
+    category = Category.objects.all()
+    context = {'latest_blog_post': latest_blog_post, 'category':category}
+    return render(request, "home.html", context)
+
+# def handle_uploaded_file(f):
```
Comment (ananthu-nair707) [inline blog/blogs/views.py:22]: Commented code removal.

File: blog/templates/post.html [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+        <a href="/blog/{{post.slug}}">{{post.title|truncatechars:70}}</a>
+    </h2>
+    <p class="lead">
+        by <a href="#">{{post.author.username}}</a>
+    </p>
+    <p>Category: <a href="#">{{post.category.title}}</a></p>
+    <p><span class="glyphicon glyphicon-time"></span> Posted on {{post.created_on}}</p>
+    <hr>
+    <img class="img-responsive" src="http://placehold.it/900x300" alt="">
+    <hr>
```
Comment (ananthu-nair707) [inline blog/templates/post.html:13]: Is this image pushed? All static files to be pushed in git. All media files to be excluded from git. Avoid remote links which can be fetched locally except for CDNs.

File: blog/blog/settings.py [hunk 5]
```diff
@@ -133,5 +133,5 @@
 EMAIL_USE_TLS = True
 EMAIL_HOST = 'smtp.gmail.com'
 EMAIL_PORT = 587
-EMAIL_HOST_USER = 'adityaagrawal99601@gmail.com'
-EMAIL_HOST_PASSWORD = 'Aaditya@9960'
```
Comment (ananthu-nair707) [inline blog/blog/settings.py]: Change your gmail password :P

**Review by ananthu-nair707 — CHANGES_REQUESTED** [review]

File: blog/blogs/forms.py (hunk 1 above)
Comment (aadityaa-cuelogic) [reply blog/blogs/forms.py]: Done.
Removed.

File: blog/blogs/urls.py (hunk 2 above)
Comment (aadityaa-cuelogic) [reply blog/blogs/urls.py]: Done.
Corrected.

File: blog/blogs/views.py (hunk 3 above)
Comment (aadityaa-cuelogic) [reply blog/blogs/views.py:22]: Its needed for future change. So kept it.

File: blog/templates/post.html (hunk 4 above)
Comment (aadityaa-cuelogic) [reply blog/templates/post.html:13]: Its a temporary link as upload image feature is not there.

**Review by aadityaa-cuelogic — COMMENTED** [review]
Kindly approve.

**Review by ananthu-nair707 — APPROVED** [review]

**Pull request merged**